sql/core/src/test/scala/org/apache/spark/sql/execution/streaming/FileStreamSinkLogSuite.scala

```diff
@@ -0,0 +1,197 @@
+    assert(Seq(5, 6, 7) === getAllValidBatches(7, compactLength = 3))
+    assert(Seq(8) === getAllValidBatches(8, compactLength = 3))
+  }
+
+  test("compactLogs") {
+    val logs = Seq(
+      FileLog("/a/b/x", 100L, FileStreamSinkLog.ADD_ACTION),
+      FileLog("/a/b/y", 200L, FileStreamSinkLog.ADD_ACTION),
+      FileLog("/a/b/z", 300L, FileStreamSinkLog.ADD_ACTION))
+    assert(logs === compactLogs(logs))
+
+    val logs2 = Seq(
+      FileLog("/a/b/m", 100L, FileStreamSinkLog.ADD_ACTION),
+      FileLog("/a/b/n", 200L, FileStreamSinkLog.ADD_ACTION),
+      FileLog("/a/b/z", 300L, FileStreamSinkLog.DELETE_ACTION))
+    assert(logs.dropRight(1) ++ logs2.dropRight(1) === compactLogs(logs ++ logs2))
+  }
+
+  test("serialize") {
+    withSQLConf(SQLConf.FILE_STREAM_SINK_LOG_COMPACT_LEN.key -> "3") {
+      withFileStreamSinkLog { sinkLog =>
+        val logs = Seq(
+          FileLog("/a/b/x", 100L, FileStreamSinkLog.ADD_ACTION),
+          FileLog("/a/b/y", 200L, FileStreamSinkLog.DELETE_ACTION),
+          FileLog("/a/b/z", 300L, FileStreamSinkLog.ADD_ACTION))
+
+        val expected = s"""${FileStreamSinkLog.VERSION}
+            |{"path":"/a/b/x","size":100,"action":"add"}
+            |{"path":"/a/b/y","size":200,"action":"delete"}
+            |{"path":"/a/b/z","size":300,"action":"add"}""".stripMargin
+        assert(expected === new String(sinkLog.serialize(logs), UTF_8))
+
+        assert(FileStreamSinkLog.VERSION === new String(sinkLog.serialize(Nil), UTF_8))
+      }
+    }
+  }
+
+  test("deserialize") {
+    withSQLConf(SQLConf.FILE_STREAM_SINK_LOG_COMPACT_LEN.key -> "3") {
+      withFileStreamSinkLog { sinkLog =>
+        val logs = s"""${FileStreamSinkLog.VERSION}
+            |{"path":"/a/b/x","size":100,"action":"add"}
+            |{"path":"/a/b/y","size":200,"action":"delete"}
+            |{"path":"/a/b/z","size":300,"action":"add"}""".stripMargin
+
+        val expected = Seq(
+          FileLog("/a/b/x", 100L, FileStreamSinkLog.ADD_ACTION),
+          FileLog("/a/b/y", 200L, FileStreamSinkLog.DELETE_ACTION),
+          FileLog("/a/b/z", 300L, FileStreamSinkLog.ADD_ACTION))
+
+        assert(expected === sinkLog.deserialize(logs.getBytes(UTF_8)))
+
+        assert(Nil === sinkLog.deserialize(FileStreamSinkLog.VERSION.getBytes(UTF_8)))
+      }
+    }
+  }
+
+  test("batchIdToPath") {
+    withSQLConf(SQLConf.FILE_STREAM_SINK_LOG_COMPACT_LEN.key -> "3") {
+      withFileStreamSinkLog { sinkLog =>
+        assert("0" === sinkLog.batchIdToPath(0).getName)
+        assert("1" === sinkLog.batchIdToPath(1).getName)
+        assert("2.compact" === sinkLog.batchIdToPath(2).getName)
+        assert("3" === sinkLog.batchIdToPath(3).getName)
+        assert("4" === sinkLog.batchIdToPath(4).getName)
+        assert("5.compact" === sinkLog.batchIdToPath(5).getName)
+      }
+    }
+  }
+
+  test("compact") {
+    withSQLConf(SQLConf.FILE_STREAM_SINK_LOG_COMPACT_LEN.key -> "3") {
```

Comment:
shouldnt this test actually verify that the logs are actually being compacted? As in *.compact files are actually being generated which contains all the contents of the previous logs.